controller.py

```diff
@@ -426,10 +428,44 @@ def _make_request(url, on_404, on_timeout, on_exception, allow_401=False):
             )
             return INVALID_INTEGRATION_DOCUMENT.detailed(failure_detail)
 
-        library, is_new = get_one_or_create(
-            self._db, Library,
-            opds_url=opds_url,
-        )
+        library = None
+        elevated_permissions = False
+        auth_url = auth_response.final_url
+        if shared_secret:
+            # Look up a library by the provided shared secret. This
+            # will let us handle the case where the library has
+            # changed URLs (auth_url does not match
+            # library.authentication_url) but the shared secret is the
+            # same.
+            library = get_one(
+                self._db, Library,
+                shared_secret=shared_secret
+            )
```

Comment:
Since we are looking up libraries by shared secret now, we might want something like the short name dupe check, even if it's extremely unlikely to be needed.